lib/ecto/repo/schema.ex

```diff
@@ -41,7 +41,9 @@ defmodule Ecto.Repo.Schema do
       |> returning(opts)
       |> fields_to_sources(dumper)
 
-    {rows, header} = extract_header_and_fields(rows, schema, dumper, autogen_id, adapter)
+    placeholder_map = Keyword.get(opts, :placeholders, %{})
+
+    {rows, header} = extract_header_and_fields(rows, schema, dumper, autogen_id, adapter, placeholder_map)
```

Comment:
Nitpick: the adapter is always the last argument so please keep it consistent. :)